src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -15,7 +17,7 @@ namespace Microsoft.CodeAnalysis
     /// Represents a diagnostic, such as a compiler error or a warning, along with the location where it occurred.
     /// </summary>
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-    public abstract partial class Diagnostic : IEquatable<Diagnostic>, IFormattable
+    public abstract partial class Diagnostic : IEquatable<Diagnostic?>, IFormattable
```

Comment:
>IEquatable<Diagnostic?> [](start = 47, length = 23)

Should this be `IEquatable<Diagnostic>` given that `IEquatable<T>` has `Equals([AllowNull]T other)`?

Reply:
`IEquatable<T>` does not have `[AllowNull]` in anything we target. The form used here has the correct outcome both before and after the annotations were added to the core library.